src/reactor.cc

```diff
@@ -0,0 +1,406 @@
+
+    cyclus::Material::Ptr reserve_fuel = tritium_reserve.Pop();
+    cyclus::Material::Ptr core_fuel = tritium_core.Pop();
+
+    core_fuel->Absorb(reserve_fuel->ExtractQty(core_deficit));
+    tritium_storage.Push(reserve_fuel->ExtractQty(surplus));
+
+    RecordOperationalInfo(
+        "Tritium Moved",
+        std::to_string(core_deficit) + "kg of T moved from reserve to core");
+    if (surplus > 0.0) {
+      RecordOperationalInfo(
+          "Tritium Moved",
+          std::to_string(surplus) + "kg of T moved from reserve to storage");
+    }
+    tritium_reserve.Push(reserve_fuel);
+    tritium_core.Push(core_fuel);
+
+    CombineInventory(tritium_storage);
+  }
+
+  // This pulls out some of the blanket each timestep so that fresh blanket can
+  // be added.
+  if (!blanket.empty() &&
+      blanket.quantity() >= blanket_size * blanket_turnover_rate) {
+    cyclus::Material::Ptr blanket_mat = blanket.Pop();
+    cyclus::Material::Ptr spent_blanket =
+        blanket_mat->ExtractQty(blanket_size * blanket_turnover_rate);
+    RecordOperationalInfo(
+        "Blanket Cycled",
+        std::to_string(spent_blanket->quantity()) + "kg of blanket removed");
+    blanket.Push(blanket_mat);
+  } else if (!blanket.empty() &&
+             blanket.quantity() < blanket_size * blanket_turnover_rate) {
+    RecordOperationalInfo(
+        "Blanket Not Cycled",
+        "Total blanket material (" + std::to_string(blanket.quantity()) +
+            ") insufficient to extract " +
+            std::to_string(blanket_size * blanket_turnover_rate) + "kg!");
+  }
+}
+
+void Reactor::Tock() {
+  if (!core_loaded) {
+    try {
+      Startup();
+      fuel_startup_policy.Stop();
+      // blanket_startup_policy.Stop();
+      fuel_refill_policy.Start();
+      core_loaded = true;
+    } catch (const std::exception& e) {
+      RecordOperationalInfo("Startup Error", e.what());
+      LOG(cyclus::LEV_INFO2, "Reactor") << e.what();
+    }
+  }
+
+  CombineInventory(tritium_reserve);
+  CombineInventory(blanket);
+
+  RecordInventories(tritium_core.quantity(), tritium_reserve.quantity(),
+                    tritium_storage.quantity(), blanket.quantity(),
+                    helium_storage.quantity());
+}
+
+void Reactor::EnterNotify() {
+  cyclus::Facility::EnterNotify();
+
+  fuel_startup_policy
+      .Init(this, &tritium_reserve, std::string("Tritium Storage"),
+            &fuel_tracker, std::string("ss"),
+            reserve_inventory + startup_inventory,
+            reserve_inventory + startup_inventory)
+      .Set(fuel_incommod)
+      .Start();
+  blanket_fill_policy
+      .Init(this, &blanket, std::string("Blanket Startup"), &blanket_tracker,
+            std::string("ss"), blanket_size, blanket_size)
+      .Set(blanket_incommod)
+      .Start();
+
+  // Tritium Buy Policy Selection:
+  if (refuel_mode == "schedule") {
+    cyclus::IntDistribution::Ptr active_dist =
+        cyclus::FixedIntDist::Ptr(new cyclus::FixedIntDist(1));
```

Comment:
I think if you're specifying a dormant dist (which is true here), it probably makes sense to specify an active dist too since they're linked, even if it is equivalent to the default. To me, it makes more sense to pass `active_dist = NULL` and `dormant_dist = NULL` when you're only specifying a buying size dist